core/domain/topic_domain.py

```diff
@@ -314,29 +330,47 @@ def create_default_story_reference(cls, story_id):
         """
         return cls(story_id, False)
 
-    def validate(self):
+    def validate(self) -> None:
         """Validates various properties of the StoryReference object.
 
         Raises:
             ValidationError. One or more attributes of the StoryReference are
                 invalid.
         """
-        if not isinstance(self.story_id, str):
+        if self.story_id == '':
             raise utils.ValidationError(
-                'Expected story id to be a string, received %s' %
-                self.story_id)
-        if not isinstance(self.story_is_published, bool):
+                'Story id should not be empty')
+
+        if not bool(re.match(constants.ENTITY_ID_REGEX, self.story_id)):
             raise utils.ValidationError(
-                'Expected story_is_published to be a boolean, received %s' %
-                self.story_is_published)
+                'Invalid story id')
```

Comment:
Move to the previous line 
```suggestion
                'Invalid story ID: %s' % self.story_id)
```

Reply:
done. please have a look and let me know is it ok or not.